libgraph: Adds fixed-binary POD array property and a view over the array

l-hoang opened the pull request:
Adds a new property type to Properties.h: FixedSizeBinaryPODArray.
This property is an array on every single node of some fixed size and is
represented with a FixedSizeBinary arrow array. Unlike the existing
ArrayProperty, it does not use arrow Lists (which add overhead to the
existing ArrayProperty since offsets are maintained even though the
ArrayProperty currently only uses fixed sized lists) and supports
arbitrary POD types. This commit also adds a view over
the arrow FixedSizeBinaryArray type assuming it contains some fixed
number of POD types.

The test this commit adds makes sure that this new view is sane.

-----

Meta-level comment:

The reason this has been added is to use it to create and persist FixedSizeBinary
arrays onto disc to use as the de-facto storage for arrays.

scober commented on libgraph/test/property-view-fixed-binary.cpp:
```diff
@@ -0,0 +1,70 @@
+
+/// Generates a fixed sized binary array for checking with a view.
+katana::Result<std::shared_ptr<arrow::FixedSizeBinaryArray>>
+GenerateTestArray() {
```
I don't know this part of the code super well, but isn't there a file for testing property views already? Does this need to be in a separate file?

l-hoang replied:
I can place it in the older test file

tylershunt reviewed (approved):
I don't know this part of the code very well, but the code makes sense to me and looks low risk since it's purely additive. So let's go!

danielmawhirter commented on libgraph/include/katana/Properties.h:
```diff
@@ -699,5 +699,126 @@ struct ArrayProperty {
+            fixed_sized_binary_builder.Finish(),
+            "failed to finish fixed size binary builder");
+
+    return katana::Result<std::shared_ptr<arrow::Table>>(arrow::Table::Make(
```
`return arrow::Table::Make(`
should work fine, there's no arrow::Result involved

danielmawhirter commented on libgraph/include/katana/Properties.h:
```diff
@@ -699,5 +699,126 @@ struct ArrayProperty {
+
+    return katana::Result<std::shared_ptr<arrow::Table>>(arrow::Table::Make(
+        arrow::schema({arrow::field(name, fixed_size_type)}),
+        {array_of_fixed_size_binaries}));
```
std::move(array...

danielmawhirter reviewed (approved):
LGTM

Commit:
libgraph: Adds fixed-binary POD array property and a view over the array

Adds a new property type to Properties.h: FixedSizeBinaryPODArray.
This property is an array on every single node of some fixed size and is
represented with a FixedSizeBinary arrow array. Unlike the existing
ArrayProperty, it does not use arrow Lists (which add overhead to the
existing ArrayProperty since offsets are maintained even though the
ArrayProperty currently only uses fixed sized lists) and supports
arbitrary POD types. This commit also adds a view over
the arrow FixedSizeBinaryArray type assuming it contains some fixed
number of POD types.

The test this commit adds makes sure that this new view is sane.
